system/jlib/jsocket.cpp

```diff
@@ -2125,15 +2125,27 @@ bool CSocket::check_connection()
         rc = sendto(sock, NULL, 0, 0, &u.sa, ul);
     }
     else {
+#ifdef _WIN32
         rc = send(sock, NULL, 0, SEND_FLAGS);
+
+#else
+        char buffer;
+        rc = recv(sock, &buffer, sizeof(buffer), MSG_PEEK | MSG_DONTWAIT);
+        if (!rc)
```

Comment:
if rc == 0 then I think you would want to return true as the socket is assumed still connected ?